### DISCUSSION TITLE
Tried installing this many times and it just hangs at the console

### DISCUSSION THREAD

**Discussion by bxlgotham** [danielmiessler/Fabric#712] [General]
I have followed the instructions to install and setup on latest MacOS. I get it all done, then run fabric and it just hangs at the console.

**Comment by patonw** [comment]
What arguments are you passing to it? In the absence of input text it tries to read from stdin, hence appearing that it hangs. Try typing in a question and ending with Ctrl+D on a new line.

**Reply by bxlgotham** [reply]
OK I tried this and get this error:

yt --transcript https://www.youtube.com/watch?v=HGS5LfyxLUU | fabric --stream --pattern extract_wisdom

zsh: no matches found: https://www.youtube.com/watch?v=HGS5LfyxLUU
Error: Error code: 404 - {'error': {'message': 'The model `gpt-4-turbo-preview` does not exist or you do not have access to it.', 'type': 'invalid_request_error', 'param': None, 'code': 'model_not_found'}}
Error code: 404 - {'error': {'message': 'The model `gpt-4-turbo-preview` does not exist or you do not have access to it.', 'type': 'invalid_request_error', 'param': None, 'code': 'model_not_found'}}

**Reply by patonw** [reply]
Several things:
- Put quotes around the URL to prevent the shell from trying to interpret it
- Do you have an `OPENAI_API_KEY` configured in your environment or `.env` file?
- Does your account/project/api key have access to gpt-4 yet? Your account needs to meet certain requirements for it.

Try `gpt-3.5-turbo` first and check the OpenAI dashboard to see if gpt-4 is selectable.

**Reply by bxlgotham** [reply]
I have the API key setup, but just created the key, how do you get 4 enabled for an API key? In the meantime how do you tell fabric to use 3.5t? Also how do I switch to Claude anyway?

**Reply by patonw** [reply]
According to their [docs](https://platform.openai.com/docs/guides/rate-limits/usage-tiers?context=tier-one) you need to have purchased $5 of credits historically, which you need to use the API at all anyhow.

You can temporarily switch models by passing the `-m model-name` flag or setting `DEFAULT_MODEL` in your .env config.

To switch to claude, you'll need to get an API key from anthropic and pass the model name of a specific claude variant to fabric. You'll also need to get credits from Anthropic to use their API too.
